typing/typemod.ml

```diff
@@ -236,7 +237,7 @@ let make_variance p n i =
 let rec iter_path_apply p ~f =
   match p with
   | Pident _ -> ()
-  | Pdot (p, _) -> iter_path_apply p ~f
+  | Pdot (p, _) | Pextra_ty (p, _) -> iter_path_apply p ~f
```

Comment:
`Pextra_ty` should be `assert false`

Reply:
Reflected at https://github.com/ocaml/ocaml/pull/11568/commits/9dfe8195f96f08462db124ab414e6701e2937882